chalk-engine/src/table.rs

```diff
@@ -116,14 +120,12 @@ impl<C: Context> Table<C> {
                 true
             }
 
-            Entry::Occupied(mut entry) => {
+            Entry::Occupied(entry) => {
                 let was_ambiguous = entry.get();
-                if !was_ambiguous || answer.ambiguous {
-                    false
-                } else {
-                    *entry.get_mut() = false;
-                    true
+                if *was_ambiguous && !answer.ambiguous {
+                    panic!("New answer was not ambiguous whereas previous answer was.");
```

Comment:
This doesn't seem like a valid reason to panic to me -- it can definitely happen that we find an ambiguous answer through one route, but a valid (unambiguous) answer through another.

Reply:
I don't quite understand why it was changed.

Reply:
I should probably create some tests to exercise this path, huh?

Reply:
That said, I think we could just ignore the answer and return `false` if we already have a better answer

Reply:
So yes, this path is never taken in any tests. I changed it to an explicit panic here because returning `true` here means that upstream the answer index gets incremented (since it's assumed `true` here means a new answer.

Reply:
So, if we want the behavior here to remain the same (that new unambiguous answers can replace ambiguous ones), then code elsewhere needs to change too.